composables/autoTeleport/useAutoTeleportTransitionDetails.ts

```diff
@@ -135,17 +138,36 @@ export default function (
     return bufferFee === 0 ? amountFee : bufferFee
   })
 
-  const amountToTeleport = computed(
+  const onlyRequiredAmountToTeleport = computed(
     () =>
       neededAmountWithFees.value +
       buffer.value -
       transferableCurrentChainBalance.value,
   )
 
-  const hasEnoughInRichestChain = computed(
-    () => (richestChainBalance.value || 0) >= amountToTeleport.value,
+  const amountToTeleport = computed(() =>
+    shouldTeleportAllBalance.value
+      ? richestChainBalance.value - (buffer.value + totalFees.value)
+      : onlyRequiredAmountToTeleport.value,
+  )
+
+  const shouldTeleportAllBalance = computed<boolean>(
+    () =>
+      Boolean(richestChain.value) &&
+      !hasEnoughInCurrentChain.value &&
+      richestChainBalance.value - onlyRequiredAmountToTeleport.value <=
+        richestChainExistentialDeposit.value,
   )
 
+  const hasEnoughInRichestChain = computed(() => {
+    const balance =
+      (shouldTeleportAllBalance.value
+        ? richestChainBalance.value
+        : transferableRichestChainBalance.value) || 0
```

Comment:
you would not need || 0 if you cap transferableRichestChainBalance to be minimun 0